.github/workflows/build-push.yml

```diff
@@ -82,22 +88,31 @@ jobs:
     - name: set up Docker Buildx
       uses: docker/setup-buildx-action@v3
     - name: Login to Docker
-      if: ${{ needs.check-secret.outputs.available == 'true' }} 
+      if: ${{ needs.check-secret.outputs.available == 'true' }}
       uses: docker/login-action@v3
       with:
           registry: ${{ inputs.image_repo }}
           username: ${{ secrets.docker_username }}
-          password: ${{ secrets.docker_password }}    
+          password: ${{ secrets.docker_password }}
+    - name: Build-push s3 image
+      if: ${{ needs.check-secret.outputs.available == 'true' && env.s3_change == 'true' }}
+      uses: docker/build-push-action@v6
+      with:
+        platforms: linux/amd64
+        push: true
```

Comment:
How's about having 
`push: ${{ inputs.push }}`
This is unlike base image. We don't have to push it for all cases. We should treat this the same as the main image.

Reply:
In which case do we not need to push? Currently it gets built-and-pushed when s3 changes.

Reply:
It is the same reason that we didn't push the PR image to the repo. If we decide to keep all PR on the repo, we should also do the same to the main image. 

Another issue I request for change is that we want to have the same s3 image tag tested in the tekton and integration test but currently it refers to the latest tag. The s3 built here is with PR tag. You have to update the tekton/tasks/s3/*.yaml to use the same image tag.

Reply:
Since build and push is run only after merge, and as the training pipeline refers to latest, the training pipeline should make use of the latest image.

Reply:
You are correct. I miss the point that we cannot test s3 on PR anyway.